src/vision/status_quo/alan_needs_async_in_traits.md

```diff
@@ -0,0 +1,101 @@
+# 😱 Status quo stories: Alan needs async in traits
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Alan is working on a project with Barbara which has already gotten off to a [somewhat rocky start](https://github.com/rust-lang/wg-async-foundations/pull/111). He is working on abstracting away the HTTP implementation the library uses so that users can provide their own. He wants the user to implement an async trait called `HttpClient` which has one method `perform(request: Request) -> Response`. Alan tries to create the async trait:
+
+```rust,ignore
+trait HttpClient {
+    async fn perform(request: Request) -> Response;
+}
+```
+
+When Alan tries to compile this, he gets an error:
+
+```ignore
+ --> src/lib.rs:2:5
+  |
+2 |     async fn perform(request: Request) -> Response;
+  |     -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  |     |
+  |     `async` because of this
+  |
+  = note: `async` trait functions are not currently supported
+  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
+```
+
+Alan, using Rust for a little while now, has learned to follow compiler error messages and adds `async-trait` to the dependencies list. Alan follows the README of `async-trait` and gets the code to compile. Alan is somewhat dismayed that the compile times have increased considerably after adding this new dependency.
```

Comment:
```suggestion
Alan, who has been using Rust for a little while now, has learned to follow compiler error messages and adds `async-trait` to his `Cargo.toml`. Alan follows the README of `async-trait`, and comes up with the following code:
'''rust
#[async_trait]
trait HttpClient {
    async fn perform(request: Request) -> Response;
}
'''
Despite getting his code to compile, Alan is somewhat dismayed that the compile times have increased considerably after adding this new dependency.
```
(markdown doesn't like nested code blocks so I used apostrophes instead of backticks - don't click "apply suggestion" :)

Reply:
+1 for having a code example.

Reply:
Maybe include @rylev's numbers:

---

```suggestion
Alan, who has been using Rust for a little while now, has learned to follow compiler error messages and adds `async-trait` to his `Cargo.toml`. Alan follows the README of `async-trait` and comes up with the following code:
'''rust
#[async_trait]
trait HttpClient {
    async fn perform(request: Request) -> Response;
}
'''
Alan's code now compiles, but he also finds that his compile times have gone from under a second to around 6s, at least for a clean build.
```